jobserver/copiloting.py

```diff
@@ -0,0 +1,52 @@
+from datetime import timedelta
+
+import structlog
+from django.conf import settings
+from django.contrib.humanize.templatetags.humanize import naturalday
+from django.utils import timezone
+from furl import furl
+
+from jobserver.models import Project
+from services.slack import client as slack_client
+
+
+logger = structlog.get_logger(__name__)
+
+
+def notify_impending_copilot_windows_closing(days=5):
+    """
+    Send a message to slack about an Application instance
```

Comment:
Nitpick: This could be more specific. What about:

> Sends a message to Slack that reports the number of closing copilot windows in the next `days` days. If there aren't any, then doesn't send a message.

Reply:
@iaindillingham – updated partly with your suggestion but also with some notes on the link style, does that explain enough now?  Happy to add more if you think it's necessary!

Reply:
It does, thank you 🙂